services/src/main/scala/cromwell/services/metadata/impl/builder/MetadataBuilderActor.scala

```diff
@@ -488,6 +515,134 @@ class MetadataBuilderActor(readMetadataWorkerMaker: () => Props,
+        val subMetadataBuilder = context.actorOf(MetadataBuilderActor.props(readMetadataWorkerMaker,
+                                                                            metadataReadRowNumberSafetyThreshold,
+                                                                            isForSubworkflows = true
+                                                 ),
+                                                 uniqueActorName(subId)
+        )
+        subMetadataBuilder ! GetCost(WorkflowId.fromString(subId))
+      }
+      goto(WaitingForSubWorkflowCost) using HasReceivedEventsData(target,
+                                                                  originalRequest,
+                                                                  metadataResponse.query,
+                                                                  metadataResponse.eventList,
+                                                                  Map.empty,
+                                                                  subWorkflowIds.size
+      )
+    }
+  }
+
+  private def extractFromJsAs[A: Manifest](js: JsObject, fieldName: String): Option[A] =
+    js.fields.get(fieldName) match {
+      case Some(a: A) => Some(a)
+      case _ => None
+    }
+
+  private def computeCost(callName: String, jsVal: JsValue): ErrorOr[BigDecimal] =
+    jsVal match {
+      case jsCall: JsObject =>
+        extractFromJsAs[JsString](jsCall, CallMetadataKeys.VmStartTime) map { startTimeVal =>
+          val startTimeErrorOr = ErrorOr(OffsetDateTime.parse(startTimeVal.value))
+
+          val endTimeErrorOr =
+            extractFromJsAs[JsString](jsCall, CallMetadataKeys.VmEndTime)
+              .map(v => ErrorOr(OffsetDateTime.parse(v.value)))
+              .getOrElse(OffsetDateTime.now().validNel)
+
+          val rawCostPerHour = extractFromJsAs[JsNumber](jsCall, CallMetadataKeys.VmCostPerHour).map(_.value)
+          val costPerHourErrorOr = rawCostPerHour match {
+            case Some(c: BigDecimal) if c >= 0 => c.validNel
+            // A costPerHour < 0 indicates an error
+            case Some(_: BigDecimal) =>
+              val index =
+                extractFromJsAs[JsNumber](jsCall, "shardIndex").map(_.value.toString).getOrElse("-1")
+              val attempt =
+                extractFromJsAs[JsNumber](jsCall, "attempt").map(_.value.toString).getOrElse("1")
+              s"Couldn't find valid vmCostPerHour for ${List(callName, index, attempt).mkString(".")}".invalidNel
+            case None => BigDecimal(0).validNel
+          }
+
+          for {
+            start <- startTimeErrorOr
+            end <- endTimeErrorOr
+            costPerHour <- costPerHourErrorOr
+            vmRuntimeInMillis = start.until(end, ChronoUnit.MILLIS).toDouble
+          } yield (vmRuntimeInMillis / (1000 * 60 * 60)) * costPerHour
+        } getOrElse (BigDecimal(0).validNel)
+      // TODO is millis the right thing to use here? Should it be seconds?
```

Comment:
millis seems fine to me. We're not going to run out, and we can (maybe) get a (slightly) more accurate estimation. FWIW the cost per hour already has many decimal places.

Reply:
I was wondering whether GCP publishes what unit they use to do their pricing calculations, have you ever come across that? Agree that seconds is a good default if not.

Reply:
They sure do. It's different for different things. You can see an example [here](https://cloud.google.com/billing/v1/how-tos/catalog-api#getting_the_list_of_skus_for_a_service). 

They mostly use cost per hour. "cost" is USD/hr, but the USD are measured down to the nano dollar. Timestamps are accurate to the nanosecond.